src/car/car.service.ts

```diff
@@ -48,51 +49,73 @@ export class CarService {
         return carUsage;
     }
 
-    async getAllCarsUsage(): Promise<object> {
+    async getAllCarsUsage(dateQueryDto?: DateQueryDto): Promise<object> {
+        const { dateFrom, dateTo } = dateQueryDto;
+
+        if (dateFrom && dateTo && dateFrom > dateTo) {
+            throw new BadRequestException('dateFrom cannot be greater than dateTo');
+        }
+        if (dateFrom && dateTo && dateFrom.getTime() === dateTo.getTime()) {
+            throw new BadRequestException('dateFrom cannot be equal to dateTo');
+        }
+        if (!dateFrom && dateTo) {
+            throw new BadRequestException('dateFrom is required');
+        } else if (dateFrom && !dateTo) {
+            throw new BadRequestException('dateTo is required');
+        }
```

Comment:
I don't like this if-else-if block
what about to set default values to dates (date.now and 01.01.1970 for example) and remove it